WCFDataService/Client/System/Data/Services/Client/ClientEdmModel.cs

```diff
@@ -495,12 +500,43 @@ private EdmTypeCacheValue GetOrCreateEdmTypeInternal(IEdmStructuredType edmBaseT
         }
 
         /// <summary>
+        /// Tries to find the backing field for property on type.
+        /// </summary>
+        /// <param name="property">The property to find the backing field for.</param>
+        /// <returns>The property backing field or null.</returns>
+        private static FieldInfo FindBackingField(PropertyInfo property, ProxyBackingFieldNamingConvention proxyBackingFieldNamingConvention)
+        {
+            FieldInfo backingField = null;
+            switch (proxyBackingFieldNamingConvention)
+            {
+                case ProxyBackingFieldNamingConvention.Auto:
+                    backingField = FindBackingField(property, ProxyBackingFieldNamingConvention.Underscores);
+                    if (backingField == null)
+                        backingField = FindBackingField(property, ProxyBackingFieldNamingConvention.CamelCasing);
+                    break;
+                case ProxyBackingFieldNamingConvention.Underscores:
+                    backingField = property.DeclaringType.GetField("__" + property.Name, BindingFlags.NonPublic | BindingFlags.Instance);
+                    if (backingField == null || backingField.FieldType != property.PropertyType)
+                        backingField = property.DeclaringType.GetField("_" + property.Name, BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Proxies may be different. The double underscores is the one I am using as the output of my RegEx example generates that. Handwritten proxies might use a single underscore syntax instead.